app/src/main/java/de/danoeh/antennapod/activity/MainActivity.java

```diff
@@ -211,7 +214,8 @@ public void onSlide(@NonNull View view, float slideOffset) {
         }
     };
 
-    public void setupToolbarToggle(@NonNull MaterialToolbar toolbar, boolean displayUpArrow) {
+    public void setupToolbarToggle(@NonNull MaterialToolbar toolbarFrag, boolean displayUpArrow) {
+        MaterialToolbar toolbar = getWeakRefToolbar(toolbarFrag);
```

Comment:
Multiple fragments can use this method at the same time. The code now assumes that there is only one. Also, how is holding a weak reference to the toolbar better than not holding any reference to it?